src/api/budget.py

```diff
@@ -7,7 +7,34 @@
 router = APIRouter()
 
 
-@router.get("/user/{user_id}/budget/{budget_category_id}", tags=["expenses"])
+def get_user(user_id: int):
+    with db.engine.connect() as conn:
+        user = conn.execute(
+            sqlalchemy.text("SELECT * FROM user WHERE user_id = :user_id"),
+            [{"user_id": user_id}]
+        ).fetchone()
+        if user is None:
+            raise HTTPException(status_code=404, detail="user not found.")
+    return user
+
+
+def get_category(user_id: int, budget_category_id: int):
+    with db.engine.connect() as conn:
+        category_user = conn.execute(
+            sqlalchemy.text('''
+            SELECT * FROM budget_category
+            WHERE user_id = :user_id
+            AND category_id = :category_id
+            '''),
+            [{"user_id": user_id, "category_id": budget_category_id}]
+        ).fetchone()
+        if category_user is None:
+            raise HTTPException(
+                status_code=404, detail="budget category not found.")
+    return category_user
+
+
+@router.get("/user/{user_id}/budget/", tags=["expenses"])
```

Comment:
Why is budget_category_id removed from the call here?

Reply:
I think this is because "by default, it will return all the user's budget information for all categories. Is this a good move? Shouldn't there be separate get_budget and list_budget apis (although we only have to do one now)

Reply:
Yeah, that was the reason I removed it. I thought it should be a query parameter if it should filter by category. But we can add a separate API for a ``list_budget`` and ``get_budget``

Reply:
We may want to change this in the future, but for now I don't mind this behavior so I'm keeping it